bootloader/src/main.c

```diff
@@ -466,9 +464,10 @@ int main(void)
 
             Reset_System();
 
-            // Initialize user application's Stack Pointer
-            __set_MSP(*(__IO uint32_t*) ApplicationAddress);
-            Jump_To_Application();
+            // Jump to system firmware
+            uint32_t addr = *(volatile uint32_t*)(ApplicationAddress + 4);
+            uint32_t stack = *(volatile uint32_t*)ApplicationAddress;
+            jump_to_system(addr, stack);
```

Comment:
For reference: this is required because GCC9 now throws a deprecated error when calling CMSIS `__set_MSP()`, which clobbers `sp`, which is in fact not what is supposed to happen.

Reply:
A quote from recent gcc docs:

> Another restriction is that the clobber list should not contain the stack pointer register. This is because the compiler requires the value of the stack pointer to be the same after an asm statement as it was on entry to the statement. However, previous versions of GCC did not enforce this rule and allowed the stack pointer to appear in the list, with unclear semantics. This behavior is deprecated and listing the stack pointer may become an error in future versions of GCC.